be/src/exprs/case_expr.cpp

```diff
@@ -155,29 +155,35 @@ class VectorizedCaseExpr final : public Expr {
                 res = ColumnHelper::cast_to_nullable_column(res);
             }
 
-            // case_column equals to when_columns[i] or not?
+            for (int i = 0; i < then_columns.size(); ++i) {
```

Comment:
use ColumnViewer instead of unpack_and_duplicate_const_column for performance.

Reply:
ColumnViewer does not support nested types.